### PR TITLE
CONSOLE-2371: [i18n] Externalize olm OperatorHub strings

### PR DESCRIPTION
**Pull request by TheRealJon**
Externalize strings in `frontend/packages/operator-lifecycle-manager/src/components/operatorhub`

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx:174]:
```suggestion
          title={t('olm~No OperatorHub items found')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx:58]: No need to localize this string.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx:81]: No need to localize this string.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx:182]: I don't think `here` will be localized.

**Review by rhamilto — APPROVED** [review]
/lgtm

/hold for approvals

/assign @yapei @ahardin-rh @sferich888

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx:200]: link text `Red Hat Marketplace` is not translated
Screen Shot 2020-12-16 at 4 51 09 PM

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:417]: Clicking 'Continue' button in `Show Community Operator` modal will close the modal instead of redirecting to operator installation page, this should be fixed
Screen Shot 2020-12-16 at 4 55 14 PM

**Comment by TheRealJon** [comment]
@yapei I've addressed you're feedback.

**Review by rhamilto — APPROVED** [review]
/lgtm

**Comment by rhamilto** [comment]
@TheRealJon rebase required.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-community-provider-modal.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-community-provider-modal.tsx:21]: Is there a reason we can't use `setIgnoreWarnings` directly? `onIgnoreChange` seems to wrap that with no other changes?

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-community-provider-modal.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-community-provider-modal.tsx:45]: Here you have "Community Operators," but you have "community Operators" in other places.

```suggestion
                'olm~Community Operators are Operators which have not been vetted or verified by Red Hat. Community Operators should be used with caution because their stability is unknown. Red Hat provides no support for community Operators.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:114]:
```suggestion
            'olm~This is a community provided Operator. These are Operators which have not been vetted or verified by Red Hat. Community Operators should be used with caution because their stability is unknown. Red Hat provides no support for community Operators.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:566]: You don't typically want to pass `t` as a prop. Not all `t` functions are the same since `useTranslation` takes parameters. Better to call `useTranslation` in each component that needs it.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-community-provider-modal.tsx
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-community-provider-modal.tsx:21]: Probably just left behind from the class component implementation. I'll fix this.

**Comment by TheRealJon** [comment]
@spadgett I've addressed your comments

**Review by spadgett — APPROVED** [review]
/lgtm

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx
Comment (yapei) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx:200]: @TheRealJon  I didn't see `Red Hat Marketplace` string is translated in new code

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (yapei) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx]: this is updated

**Comment by yapei** [comment]
@TheRealJon  I see one minor issue: `Red Hat Marketplace` string is not translated in new code, otherwise everything looks good to me

**Comment by TheRealJon** [comment]
@spadgett rebased. Needs lgtm again.

**Comment by TheRealJon** [comment]
@yapei This is ready for QE review.

**Comment by TheRealJon** [comment]
@yapei I addressed [your comment](https://github.com/openshift/console/pull/7501#issuecomment-755854098). This should be ready for QE review again.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by yapei** [comment]
Thank you @TheRealJon  now everything looks good

**Review by spadgett — APPROVED** [review]
/lgtm

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**
